### ISSUE TITLE
POLARS_MAX_THREADS does not seem to work

### ISSUE THREAD

**Issue by jzombi** [pola-rs/polars#5627]
### Polars version checks

- [x] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

According to the [documentation of the rust package](https://docs.rs/polars/latest/polars/#config-with-env-vars) and [comment](https://github.com/pola-rs/polars/issues/1718#issuecomment-964238805) on github issue, the `POLARS_MAX_THREADS` environment variable should restrict the number of threads used by polars.

I could not get it to work. Am I doing something wrong?

I've tested it with different polars versions (0.10, 0.11, 0.12, 0.13, 0.14), but could not find one that respects the setting.

### Reproducible example

```python
#!/usr/bin/env python
# Usage:
#  $ POLARS_MAX_THREADS=1 python max_threads.py

import os
import psutil
import time
import numpy as np
print('POLARS_MAX_THREADS={}'.format(os.environ['POLARS_MAX_THREADS']))
import polars as pl

df = pl.DataFrame({'x': np.random.rand(10_000_000)})

print('Num threads before: {}'.format(psutil.Process().num_threads()))

cpu_t0 = psutil.Process().cpu_times()
t0 = time.time()
for _ in range(10):
    df.sort('x')
t1 = time.time()
cpu_t1 = psutil.Process().cpu_times()

print('Wall Time: {:.2f}s'.format(t1-t0))
print('CPU time: {:.2f}s'.format(cpu_t1.user + cpu_t1.system - cpu_t0.user - cpu_t1.system))
print('Num threads after: {}'.format(psutil.Process().num_threads()))

# Output on my 2 core HT CPU:

# POLARS_MAX_THREADS=1
# Num threads before: 1
# Wall Time: 7.02s
# CPU time: 19.62s
# Num threads after: 5
```
(Edit: fixed psutil usage to include the cpu usage for the actual process only.)

### Expected behavior

Num threads is expected to be 1 (or 2?) even after sort, CPU time should be <= Wall time.

### Installed versions

<details>

```
---Version info---
Polars: 0.14.31
Index type: UInt32
Platform: Linux-5.19.12-arch1-1-x86_64-with-glibc2.36
Python: 3.10.7 (main, Sep  6 2022, 21:22:27) [GCC 12.2.0]
---Optional dependencies---
pyarrow: 9.0.0
pandas: 1.4.0
numpy: 1.21.5
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: <not installed>
matplotlib: 3.5.1
```

</details>

**Comment by aldanor** [comment]
@phaile2 have you considered opening a PR with your fixes?

**Comment by ghuls** [comment]
Most of the threads you see are due to openblas or intel MKL when you import numpy:

```python
❯ POLARS_MAX_THREADS=1 ipython
Python 3.8.10 | packaged by conda-forge | (default, May 11 2021, 07:01:05) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.11.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import psutil

In [2]: print('Num threads before: {}'.format(psutil.Process().num_threads()))
Num threads before: 2

In [3]: import numpy as np

In [4]: print('Num threads before: {}'.format(psutil.Process().num_threads()))
Num threads before: 13

```

After restricting the number of OPENBLAS threads:
```python
❯ POLARS_MAX_THREADS=1 OMP_NUM_THREADS=1 ipython
Python 3.8.10 | packaged by conda-forge | (default, May 11 2021, 07:01:05) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.11.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import psutil

In [2]: print('Num threads before: {}'.format(psutil.Process().num_threads()))
Num threads before: 2

In [3]: import numpy as np

In [4]: print('Num threads before: {}'.format(psutil.Process().num_threads()))
Num threads before: 2

In [5]: import polars as pl

In [6]: print('Num threads before: {}'.format(psutil.Process().num_threads()))
Num threads before: 3
```

**Comment by jzombi** [comment]
@ghuls in my case no:

create_df.py:
```
import numpy as np
import polars as pl

df = pl.DataFrame({'x': np.random.rand(10_000_000)})
df.write_csv('data.csv')
```

max_threads_without_numpy.py
```
import os
import psutil
import time
print('POLARS_MAX_THREADS={}'.format(os.environ['POLARS_MAX_THREADS']))
import polars as pl

df = pl.read_csv('data.csv')

print('Num threads before: {}'.format(psutil.Process().num_threads()))

cpu_t0 = psutil.Process().cpu_times()
t0 = time.time()
for _ in range(10):
    df.sort('x')
t1 = time.time()
cpu_t1 = psutil.Process().cpu_times()

print('Wall Time: {:.2f}s'.format(t1-t0))
print('CPU time: {:.2f}s'.format(cpu_t1.user + cpu_t1.system - cpu_t0.user - cpu_t1.system))
print('Num threads after: {}'.format(psutil.Process().num_threads()))
```

Results:
```
$ python create_df.py 
$ POLARS_MAX_THREADS=1 OMP_NUM_THREADS=1 python max_threads_without_numpy.py 
POLARS_MAX_THREADS=1
Num threads before: 2
Wall Time: 5.43s
CPU time: 19.21s
Num threads after: 6
```

**Comment by ghuls** [comment]
@jzombi What is the output of:
```
import polars as pl
import numpy as np

print(pl.show_versions())
print(np.show_config())
```

**Comment by jzombi** [comment]
<details>
<summary>print(pl.show_versions())</summary>

```
---Version info---
Polars: 0.14.31
Index type: UInt32
Platform: Linux-5.19.12-arch1-1-x86_64-with-glibc2.36
Python: 3.10.7 (main, Sep  6 2022, 21:22:27) [GCC 12.2.0]
---Optional dependencies---
pyarrow: 11.0.0
pandas: 1.4.0
numpy: 1.21.5
fsspec: 2023.1.0
connectorx: <not installed>
xlsx2csv: <not installed>
matplotlib: 3.5.1
None
```

</details>

<details>
<summary>print(np.show_config())</summary>

```
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
blas_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
lapack_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
Supported SIMD extensions in this NumPy install:
    baseline = SSE,SSE2,SSE3
    found = SSSE3,SSE41,POPCNT,SSE42,AVX,F16C,FMA3,AVX2
    not found = AVX512F,AVX512CD,AVX512_KNL,AVX512_KNM,AVX512_SKX,AVX512_CLX,AVX512_CNL,AVX512_ICL
None
```

</details>

**Comment by ghuls** [comment]
@jzombi You are running an old version of Polars.

**Comment by jzombi** [comment]
@ghuls  yes, and it seems to be fixed in the newest? (0.16.11) release :tada:, thanks.
```
$ POLARS_MAX_THREADS=1 OMP_NUM_THREADS=1 python max_threads_without_numpy.py 
POLARS_MAX_THREADS=1
Num threads before: 3
Wall Time: 12.37s
CPU time: 11.95s
Num threads after: 3

$ POLARS_MAX_THREADS=1 python max_threads_without_numpy.py 
POLARS_MAX_THREADS=1
Num threads before: 6
Wall Time: 11.95s
CPU time: 11.53s
Num threads after: 6
```
